src/annotation/askui-annotation-step-reporter.ts

```diff
@@ -0,0 +1,80 @@
+import fs from "fs";
+import { Step, Reporter, Annotation, ReporterConfig } from "askui";
+import path from "path";
+import { JSDOM } from 'jsdom';
+
+export enum AnnotationLevel {
+  DISABLED = 'disabled',
+  ON_FAILURE = 'onFailure',
+  ALL = 'all',
+}
+
+export interface AnnotationReporterConfig extends ReporterConfig {
+  annotationLevel: AnnotationLevel;
+  folderPath: string;
+  fileNameTemplate: string;
```

Comment:
Make them optional if you have defaults that the user can use:
```suggestion
  annotationLevel?: AnnotationLevel;
  folderPath?: string;
  fileNameTemplate?: string;
```